core/binding/src/sdk/mod.rs

```diff
@@ -0,0 +1,161 @@
+pub mod chain_querier;
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use protocol::fixed_codec::FixedCodec;
+use protocol::traits::{
+    ChainQuerier, RequestContext, ServiceSDK, ServiceState, StoreArray, StoreBool, StoreMap,
+    StoreString, StoreUint64,
+};
+use protocol::types::{Address, Epoch, Hash, Receipt, SignedTransaction};
+use protocol::ProtocolResult;
+
+use crate::store::{
+    DefaultStoreArray, DefaultStoreBool, DefaultStoreMap, DefaultStoreString, DefaultStoreUint64,
+};
+
+pub struct DefalutServiceSDK<S: ServiceState, C: ChainQuerier, R: RequestContext> {
+    state:           Rc<RefCell<S>>,
+    chain_querier:   Rc<C>,
+    request_context: R,
+}
+
+impl<S: ServiceState, C: ChainQuerier, R: RequestContext> DefalutServiceSDK<S, C, R> {
+    pub fn new(state: Rc<RefCell<S>>, chain_querier: Rc<C>, request_context: R) -> Self {
+        Self {
+            state,
+            chain_querier,
+            request_context,
+        }
+    }
+}
+
+impl<S: 'static + ServiceState, C: ChainQuerier, R: RequestContext> ServiceSDK
+    for DefalutServiceSDK<S, C, R>
+{
+    type ContextItem = R;
```

Comment:
unnecessary

Reply:
I think it's necessary.
```rust
pub trait ServiceSDK {
    type ContextItem: RequestContext;

```
return type of `fn get_request_context` is a generic type implementing trait RequestContext

Reply:
ok, isee